alf/algorithms/muzero_algorithm.py

```diff
@@ -455,10 +455,22 @@ def _unfold1_adapting_episode_ends(x):
             if type(self._data_transformer) == IdentityDataTransformer:
                 observation = replay_buffer.get_field(
                     'observation', folded_env_ids, folded_positions)
+                observation = _unfold1_adapting_episode_ends(observation)
             else:
+                # In contrast to the preceding case, where we can first extract
+                # observation and then unfold it, certain data transformers,
+                # such as FrameStacker, are sensitive to the order in which the
+                # unfold is applied. As a result, we choose to first unfold.
                 observation, step_type = replay_buffer.get_field(
                     ('observation', 'step_type'), folded_env_ids,
                     folded_positions)
+
+                # Unfold (and reshape)
+                observation = _unfold1_adapting_episode_ends(observation)
+                observation = observation.reshape(-1, *observation.shape[2:])
+                step_type = _unfold1_adapting_episode_ends(step_type)
+                step_type = step_type.reshape(-1, *step_type.shape[2:])
```

Comment:
with this, batch_info.positions should also be updated before calling transform_experience

Reply:
Yep I realized this after running the `mini_batch_length > 1` experiment. Will put up a fix.

Reply:
It is fixed.